pixeltable/functions/pil/image.py

```diff
@@ -55,73 +46,62 @@ def _crop_return_type(bound_args: Dict[str, Any]) -> ColumnType:
-getcolors = func.make_library_function(
-    JsonType(), [ImageType(), IntType()], 'PIL.Image', 'Image.getcolors')
-getextrema = func.make_library_function(JsonType(), [ImageType()], 'PIL.Image', 'Image.getextrema')
-getpalette = \
-    func.make_library_function(JsonType(), [ImageType(), StringType()], 'PIL.Image', 'Image.getpalette')
-getpixel = func.make_library_function(
-    JsonType(), [ImageType(), ArrayType((2,), dtype=IntType())], 'PIL.Image', 'Image.getpixel')
-getprojection = func.make_library_function(JsonType(), [ImageType()], 'PIL.Image', 'Image.getprojection')
-quantize = func.make_library_function(
-    ImageType(), [ImageType(), IntType(), IntType(nullable=True), IntType(), IntType(nullable=True), IntType()],
-    'PIL.Image', 'Image.quantize')
-reduce = \
-    func.make_library_function(ImageType(), [ImageType(), IntType(), JsonType()], 'PIL.Image', 'Image.reduce')
-transpose = func.make_library_function(
-    _caller_return_type, [ImageType(), IntType()], 'PIL.Image', 'Image.transpose')
 
-func.FunctionRegistry.get().register_module(sys.modules[__name__])
+effect_spread = func.make_module_function(
+    PIL.Image.Image.effect_spread, return_type=_caller_return_type, param_types=[ImageType(), FloatType()])
+entropy = func.make_module_function(
+    PIL.Image.Image.entropy, return_type=FloatType(), param_types=[ImageType(), ImageType(), JsonType()])
+getbands = func.make_module_function(PIL.Image.Image.getbands, return_type=JsonType(), param_types=[ImageType()])
+getbbox = func.make_module_function(PIL.Image.Image.getbbox, return_type=JsonType(), param_types=[BoolType()])
+getcolors = func.make_module_function(
+    PIL.Image.Image.getcolors, return_type=JsonType(), param_types=[ImageType(), IntType()])
+getextrema = func.make_module_function(PIL.Image.Image.getextrema, return_type=JsonType(), param_types=[ImageType()])
+getpalette = func.make_module_function(
+    PIL.Image.Image.getpalette, return_type=JsonType(), param_types=[ImageType(), StringType()])
+getpixel = func.make_module_function(
+    PIL.Image.Image.getpixel, return_type=JsonType(), param_types=[ImageType(), ArrayType((2,), dtype=IntType())])
+getprojection = func.make_module_function(
+    PIL.Image.Image.getprojection, return_type=JsonType(), param_types=[ImageType()])
+histogram = func.make_module_function(
+    PIL.Image.Image.histogram, return_type=JsonType(), param_types=[ImageType(), ImageType(), JsonType()])
+quantize = func.make_module_function(
+    PIL.Image.Image.quantize, return_type=ImageType(),
+    param_types=[ImageType(), IntType(), IntType(nullable=True), IntType(), IntType(nullable=True), IntType()])
+reduce = func.make_module_function(
+    PIL.Image.Image.reduce, return_type=ImageType(), param_types=[ImageType(), IntType(), JsonType()])
+transpose = func.make_module_function(
+    PIL.Image.Image.transpose, return_type=_caller_return_type, param_types=[ImageType(), IntType()])
```

Comment:
I have to say I don't love this pattern. In many of these cases, the relevant function can be replaced by a use of `apply`, e.g.,

`t.img.getprojection`

is identical in effect to

`t.img.apply(PIL.Image.Image.getprojection)`

We should think through our general philosophy regarding the "pixeltable library". If our general design pattern is to explicitly define such operations in the pixeltable library, we risk ending up with a large, sprawling stdlib that we have to maintain, one that interposes a new library the users have to come to terms with, and that will always be incomplete. An alternative approach is to use `apply` for these "nuts and bolts" operations that end up being no-op delegations to the underlying python library, and (with perhaps a few exceptions) reserve explicit pixeltable library declarations for AI related functionality.

I also don't love the make_* pattern. I'd much prefer to see all operations defined with decorators, using a common pattern.